jmespath/jputil/util.go

```diff
@@ -0,0 +1,165 @@
+package jputil
+
+import (
+	"errors"
+	"reflect"
+)
+
+// IsFalse determines if an object is false based on the JMESPath spec.
+// JMESPath defines false values to be any of:
+// - An empty string array, or hash.
+// - The boolean value false.
+// - nil
+func IsFalse(value interface{}) bool {
```

Comment:
This is a great case for using a switch statement instead of if else

```go
switch v := value.(type) {
case bool:
	return !v
case []interface{}:
	return len(v) == 0
case map[string]interface{}:
    return len(v) == 0
case string:
    return len(v) == 0
}
return false
```